x-pack/elastic-agent/pkg/agent/application/upgrade/step_download.go

```diff
@@ -20,7 +22,17 @@ import (
 	"github.com/elastic/beats/v7/x-pack/elastic-agent/pkg/release"
 )
 
-func (u *Upgrader) downloadArtifact(ctx context.Context, version, sourceURI string) (string, error) {
+func (u *Upgrader) downloadArtifact(ctx context.Context, version, sourceURI string) (_ string, err error) {
+	// TODO: span type?
+	span, ctx := apm.StartSpan(ctx, "Upgrader.downloadArtifact()", "custom")
```

Comment:
I think the span type should be "app.internal" (i.e span type = app, span.subtype = internal), according to https://github.com/elastic/apm/pull/513. @SylvainJuge is that correct?